### PR TITLE
Activate `INode::reset()`

### PR DESCRIPTION
**Pull request by brettviren**
This is for issue #30.

I tested this with my ZPB test for both `TbbFlow` and `Pgrapher` as well as the units tests.  

Please take a look and consider if this extension is not going to cause any future problems.

### CODE REVIEW

**Comment by brettviren** [comment]
@HaiwangYu thanks for the discussion.  Let me work on this more.

My thinking is that a `reset()` "means" that a component may go back to its pre-configured state.  And I see now that this PR would be bad for this meaning when WCT is run from larsoft as `reset()` would be called each "event".

**Comment by brettviren** [comment]
This adds a new `ITerminal::finalize()` method which is called from `Main::finalize()` on all known components that implement.  `Main` calls this in its destructor.  This should preserve same behavior for both `wire-cell` CLI and `WireCellToolkit_module` in art.  The `reset()` function goes back to being largely vestigial.

File: pgraph/inc/WireCellPgraph/Pgrapher.h [hunk 1]
```diff
@@ -72,6 +73,7 @@ namespace WireCell {
             virtual WireCell::Configuration default_configuration() const;
         private:
             Graph m_graph;
+            Pgraph::Factory m_fac;
```
Comment (HaiwangYu) [inline pgraph/inc/WireCellPgraph/Pgrapher.h]: Seems `m_fac` is not used in the latest commit

Comment (brettviren) [reply pgraph/inc/WireCellPgraph/Pgrapher.h]: Thanks.  Latest update removes it.

**Comment by HaiwangYu** [comment]
I compiled the code and tested with [this jsonnet](https://github.com/HaiwangYu/wct-analysis/blob/master/exp_data/dec-to-sig.jsonnet) using `Pgrapher` and `TbbFlow` in larsoft.
Results looks good.

**Pull request merged**